core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3148,7 +3148,7 @@ private void checkPushJoinThroughUnionOnRightDoesNotMatchSemiOrAntiJoin(JoinRelT
     sql(sql)
         .withRule(CoreRules.UNION_TO_VALUES)
         .withInSubQueryThreshold(0)
-        .check();
+        .checkUnchanged();
```

Comment:
it really looks like these tests are no longer testing this particular rule.
I am not sure what should be done about that.

Reply:
I would optimize this part, for setting  simplifyValues to false we can do this optimization, and also for that reason I keep this rule